### PR TITLE
Allow to use ramsey/uuid 3.0

### PR DESCRIPTION
**Pull request by MattKetmo**

### CODE REVIEW

**Comment by MattKetmo** [comment]
I also removed php 5.3 from travis because ramsey/uuid 3.0 is php >= 5.4.
But if you prefer I can keep php 5.3 and use ramsey/uuid 2.x.

**Comment by wjzijderveld** [comment]
Hmmm, not sure about this.

What is the reason you want to make it optional?

**Comment by MattKetmo** [comment]
I did not make nothing optional in this PR, you probably wanted to comment about qandidate-labs/broadway/pull/220.

If yes, I just make it "non-required" by default because `DBALEventStore` is not a mandatory component of the Broadway lib (it's just one implementation among others of one of the broadway services). The `broadway-uuid-generator` package is here to abstract the need of a specific library.

But it's not the main purpose of the PR. The goal is to be able to use ramsey/uuid 3.x (and 2.x for BC).
If you prefer I can update the composer.json to something like `"ramsey/uuid": "^2.0|^3.0"` and keep it in the `require` section.

**Comment by MattKetmo** [comment]
Up on this PR. I kept the ability to use the 2.\* version of ramsey/uuid so there should not be a reason to not merge it :)

File: composer.json [hunk 1]
```diff
@@ -5,19 +5,19 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "require-dev": {
-        "rhumsaa/uuid": "~2.4"
+        "ramsey/uuid": "^2.0|^3.0"
```
Comment (moleculezz) [inline composer.json:8]: Should be `||`
https://getcomposer.org/doc/articles/versions.md#range

EDIT:
Also, you should probably update the README

**Comment by othillo** [comment]
@MattKetmo thanks for this PR. I elaborated on your work in https://github.com/qandidate-labs/broadway-uuid-generator/pull/8 to have the class_exists call only occur during object instantiation.

**Pull request closed without merging**